app/routines/update_user_salesforce_info.rb

```diff
@@ -191,6 +191,13 @@ def cache_contact_data_in_user!(contact, user)
 
       user.faculty_status = case contact.faculty_verified
       when "Confirmed"
+        SecurityLog.create!(
+            user: user.try(:is_anonymous?) ? nil : user,
```

Comment:
I think the anonymous check is not needed here, the user should never be anonymous if they are getting faculty-verified. You should be able to just pass in the user directly.